File: tools/core/coords.mjs
// Node accessor for the shared sprite-coordinate contract. The values live in
// src/render/coords.json (single source — the browser imports it directly and
// tools/qa/coords.py reads it for Python). Import from here in gen scripts and
// the vite middleware; never re-declare these constants.
import { readFileSync } from 'node:fs';
import { fileURLToPath } from 'node:url';
import { dirname, join } from 'node:path';

const ROOT = join(dirname(fileURLToPath(import.meta.url)), '..', '..');
const c = JSON.parse(readFileSync(join(ROOT, 'src', 'render', 'coords.json'), 'utf8'));

export const CELL_W = c.cellW;
export const CELL_H = c.cellH;
export const FLOOR_FRAC = c.floorFrac;
export const HEADROOM = c.headroom;
export const ART_MARGIN = c.artMargin;
export const SPRITE_FOOT_OFFSET_Y = c.spriteFootOffsetY;
export const ORIGIN_CX = CELL_W / 2;
export const ORIGIN_FEET = FLOOR_FRAC * CELL_H;
export const CHROMA_GREEN = c.chromaGreen;
export const CHROMA_MAGENTA = c.chromaMagenta;

File: tools/qa/coords.py
"""Python accessor for the shared sprite-coordinate contract.

The values live in src/render/coords.json (single source — the browser and
tools/core/coords.mjs read the same file). Import from here in QA scripts;
never re-declare these constants.
"""
import json
import os

_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", ".."))
with open(os.path.join(_ROOT, "src", "render", "coords.json")) as _f:
    _c = json.load(_f)

CELL_W = _c["cellW"]
CELL_H = _c["cellH"]
FLOOR_FRAC = _c["floorFrac"]
HEADROOM = _c["headroom"]
ART_MARGIN = _c["artMargin"]
SPRITE_FOOT_OFFSET_Y = _c["spriteFootOffsetY"]
ORIGIN_CX = CELL_W / 2
ORIGIN_FEET = FLOOR_FRAC * CELL_H
CHROMA_GREEN = _c["chromaGreen"]
CHROMA_MAGENTA = _c["chromaMagenta"]
